src/backend/distributed/commands/cascade_table_operation_for_connected_relations.c

```diff
@@ -72,7 +69,28 @@ CascadeOperationForConnectedRelations(Oid relationId, LOCKMODE lockMode,
 		return;
 	}
 
-	LockRelationsWithLockMode(fKeyConnectedRelationIdList, lockMode);
+	CascadeOperationForRelationIdList(fKeyConnectedRelationIdList, lockMode,
+									  cascadeOperationType);
+}
+
+
+/*
+ * CascadeOperationForRelationIdList executes citus table function specified
+ * by CascadeOperationType argument on each relation in the relationIdList;
+ * Also see CascadeOperationType enum definition for supported
+ * citus table functions.
+ */
+void
+CascadeOperationForRelationIdList(List *relationIdList, LOCKMODE lockMode,
+								  CascadeOperationType
+								  cascadeOperationType)
+{
+	LockRelationsWithLockMode(relationIdList, lockMode);
+
+	if (cascadeOperationType == CASCADE_ADD_LOCAL_TABLE_TO_METADATA)
+	{
+		ErrorIfConvertingMultiLevelPartitionedTable(relationIdList);
+	}
```

Comment:
In `CreateCitusLocalTable` function, this check would never error out, since `CreateCitusLocalTable` gets called with partition relations, **after** they are detached.

Instead, here, it would error out if the user tries to convert a multi-level table, since partitioned table convertions always go through here.

Reply:
I was actually thinking if we can move the partitioning logic down to CreateCitusLocalTable itself, but now I got the reason in our teams chat discussion, so that's ok then.

Reply:
But still, can we put some code-comments to state the reason you shared here ?